pynxos/device.py

```diff
@@ -21,7 +22,12 @@ def __init__(self, host, username, password, transport=u'http', port=None, timeo
         self.timeout = timeout
         self.verify = verify
 
-        self.rpc = RPCClient(host, username, password, transport=transport, port=port, verify=self.verify)
```

Comment:
I would rather require UDS connection be explicit than assumed when a password is not sent. Probably best to add a `connection_type` argument that defaults to `http` and also accepts `uds`. I could also see having a `uds` argument that takes a boolean, but I am thinking to former option makes more sense.